src/lib.rs

```diff
@@ -0,0 +1,141 @@
+extern crate diesel;
+extern crate byteorder;
+extern crate geo;
+
+use byteorder::{ByteOrder, LittleEndian, BigEndian};
+use std::error::Error;
+use std::io::prelude::*;
+use diesel::pg::{Pg, PgTypeMetadata};
+use diesel::query_builder::QueryId;
+use diesel::row;
+use diesel::types::{FromSql, ToSql, IsNull, HasSqlType, FromSqlRow};
+use std::fmt;
+use geo::*;
+
+
+pub struct Geography {
+    pub big_endian: bool,
+    pub srid: Option<u32>,
+    pub geom: geo::Geometry<f64>,
+}
+
+#[derive(Debug)]
+struct NotImplemented {}
+impl fmt::Display for NotImplemented {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Not implemented Error")
+    }
+}
+
+impl Error for NotImplemented {
+    fn description(&self) -> &str {
+        "Not implemented"
+    }
+}
+
+impl Geography {
+    pub fn from(bytes: &[u8]) -> Result<Self, Box<Error + Send + Sync>> {
+        let big_endian = bytes[0] == 0u8;
+        let type_id = read_u32(&bytes[1..5], big_endian);
+        let srid = if type_id & 0x20000000 == 0x20000000 {
+            Some(read_u32(&bytes[5..9], big_endian))
+        } else {
+            None
+        };
+
+        match type_id & 0xFF {
+            0x01 => {
+                Ok(Geography {
+                    big_endian: big_endian,
+                    srid: srid,
+                    geom: Geometry::Point(Point::<f64>::new(read_f64(&bytes[9..17], big_endian),
+                                                            read_f64(&bytes[17..25], big_endian))),
+                })
+            }
+            _ => Err(Box::new(NotImplemented {})),
+        }
+    }
+}
+
+fn read_u32(bytes: &[u8], big_endian: bool) -> u32 {
```

Comment:
this signature is clear, but I think it's error prone, I would have taken the index as a parameter, and created the slice with the right range (4 for u32, 8 for f64 in the function)